### DISCUSSION TITLE
type PageComponent = (props: PageProps) => React.ReactNode | Promise<React.ReactNode>

### DISCUSSION THREAD

**Discussion by sylvain-guehria** [vercel/next.js#43046] [Help]
### Verify canary release

- [X] I verified that the issue exists in the latest Next.js canary release

### Provide environment information

When running tsc --noEmit in a basic project where I have a root layout in the /app folder and other sub folder with their own page and layout : 

![image](https://user-images.githubusercontent.com/51234823/202450316-2c67ade0-5cee-4486-a914-ebbb42c2cc4c.png)

I got an error on each sub layouts and pages. 

type PageComponent = (props: PageProps) => React.ReactNode | Promise<React.ReactNode>

Type error: 'PageComponent' is declared but never used.

type LayoutComponent = (props: LayoutProps) => React.ReactNode | Promise<React.ReactNode>

type LayoutComponent = (props: LayoutProps) => React.ReactNode | Promise<React.ReactNode>

![image](https://user-images.githubusercontent.com/51234823/202451478-c925c93c-8806-4414-8ad1-2e40d1ba101e.png)

### Describe the Bug

next tsc --noEmit finds typescript error that doesn't seem not exist

### Expected Behavior

No tsc error

### Link to reproduction - Issues with a link to complete (but minimal) reproduction code will be addressed faster

tsc --noEmit 

### To Reproduce

tsc --noEmit

**Comment by icyJoseph** [comment]
Hi,

This issue happens because TS expects children to be JSX elements, while the TS is fixed, you can just add:

```tsx
/* @ts-expect-error Server Component */
<MyServerComponent />
```

Once TS starts to understand Server Components, it'll complain that the `expect-error` directive is unused, and you'll be able to quickly clean up your app.

I ran `tsc --noEmit` on an out of the box project, as well as other and no hit.
